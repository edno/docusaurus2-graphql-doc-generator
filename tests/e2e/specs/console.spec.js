const exec = require("child_process").exec;

test("Code should be 0", async () => {
  const result = await cli();
  expect(result.code).toBe(0);
});

function cli(args = []) {
  return new Promise((resolve) => {
    exec(
      `npx docusaurus graphql-to-doc ${args.join(" ")}`,
      { cwd: process.cwd() },
      (error, stdout, stderr) => {
        resolve({
          code: error && error.code ? error.code : 0,
          error,
          stdout,
          stderr,
        });
      },
    );
  });
}
