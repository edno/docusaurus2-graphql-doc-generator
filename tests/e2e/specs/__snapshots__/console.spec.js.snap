// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql-to-doc should return 0 with success message when completed 1`] = `
Object {
  "code": 0,
  "error": null,
  "stderr": "",
  "stdout": "Documentation successfully generated in \\"docs\\" with base URL \\".\\".
229 pages generated in 3.508s from schema \\"https://graphql.anilist.co/\\".
Remember to update your Docusaurus site's sidebars with \\"docs/sidebar-schema.js\\".
",
}
`;
